pkg/rtc/participant.go

```diff
@@ -701,7 +701,13 @@ func (p *ParticipantImpl) GetConnectionQuality() livekit.ConnectionQuality {
 		if subTrack.IsMuted() {
 			continue
 		}
-		if subTrack.DownTrack().TargetSpatialLayer() < subTrack.DownTrack().MaxSpatialLayer() {
+		// determine the max spatial layer based on what's published
```

Comment:
Would be good to tuck all this away in DownTrack. I believe I have some of this already there in the latest. Will look at it when doing the next version of stream allocator as that is going to shake up things a fair bit.

Reply:
sounds good.